AutoSitemap_body.php

```diff
@@ -105,20 +105,26 @@ static public function writeSitemap() {
         $count = $res->numRows();
         $pos   = 0;
 
-        $data = $wgAutoSitemap["header"];
+        self::write($file_handle, $wgAutoSitemap["header"]);
         while($row = $res->fetchObject()) {
-            $data .= self::formatResult($server, $row, $pos, $count);
+            self::write($file_handle, self::formatResult($server, $row, $pos, $count));
             ++$pos;
         }
-        $data .= $wgAutoSitemap["footer"];
+        self::write($file_handle, $wgAutoSitemap["footer"]);
 
-        fwrite($file_handle, utf8_encode($data));
         fclose($file_handle);
         rename($tmp_filename, $filename);
 
         self::notifySitemap();
     }
 
+    static function write($handle, $data) {
+        $retval = fwrite($handle, $data);
+        if ($retval === FALSE || $retval === 0) {
+            die('Error while writing data.');
```

Comment:
Crashing the entire php process because of the issue while generating new sitemap does not seem okay for me

Reply:
Fixed.  Note that this was also an issue on Line 100, before my changes, which I've also fixed in the same commit.